spec/views/application/_navbar.html.erb_spec.rb

```diff
@@ -2,27 +2,43 @@
 
 require 'rails_helper'
 RSpec.describe 'application/_navbar.html.erb', type: :view do
-  let(:current_user) { FactoryBot.create :user }
+  context 'the current user is not an admin' do
+    let(:current_user) { FactoryBot.create :user }
 
-  before do
-    sign_in current_user
-    allow(view).to receive(:current_user).and_return(current_user)
-    render
-  end
+    before do
```

Comment:
You could pull out the before do block in both contexts (its the same code) and just set different current_users

describe....

  before do
    .... 
  end

  context
    let(:current_user) not admin
  end

  context
    let(:current_user)...
  end

Reply:
Nice, doesn't know that would work.